pdm/cli/actions.py

```diff
@@ -557,11 +558,39 @@ def print_pep582_command(ui: termui.UI, shell: str = "AUTO"):
         shell = shellingham.detect_shell()[0]
     shell = shell.lower()
     if shell in ("zsh", "bash"):
-        result = f"export PYTHONPATH='{lib_path}':$PYTHONPATH"
+        result = textwrap.dedent(
+            f"""
+            if [ -n "$PYTHONPATH" ]; then
+                export PYTHONPATH='{lib_path}':$PYTHONPATH
+            else
+                export PYTHONPATH='{lib_path}'
+            fi
+            """
+        ).strip()
```

Comment:
I think this would be better, with the downside of not being that readable maybe though!?
`result = f"export PYTHONPATH='{lib_path}'${{PYTHONPATH:+:${{PYTHONPATH}}}}"`

(ref: https://unix.stackexchange.com/questions/26047/how-to-correctly-add-a-path-to-path)

Reply:
> Apart from improving it for bash/zsh I think it could get improved in general by looking at `os.environ` to have the `if` / branches inside of Python (`pdm --pep582`) instead of in the shell code (which is meant to be evaluated right away, and should get an exported `$PYTHONPATH` then).

Using the shell code instead of executing `pdm --pep582` takes less overhead of shell startup.

------

Using shell parameter expansion in bash and zsh will get the shell code like below.

`export PYTHONPATH=/.../pdm/pep582${PYTHONPATH:+:${PYTHONPATH}} `

I think the readability of this line of code is ok to me. But it should be simpler and more readable for users due to that the output of `pdm --pep582` is an example to show how to enable the PEP582 automatically. Users always can alter the result depending on what they want.

Reply:
> I think it could get improved in general by looking at os.environ to have the if / branches inside of Python

The determine of which branch to take shouldn't happen when running `pdm --pep582`, as the output is often stored in rc files and other env var manipulation may take place before that.